### ISSUE TITLE
OT 1.5 (stable) freezes when rendering preview of the entire project

### ISSUE THREAD

**Issue by Al532** [opentoonz/opentoonz#3916]
My project is 1946 frames long. If I try to render a preview of the entire project, it freezes: OT becomes idle in Windows task manager and I have to force quit. It happens at a random frame (about the 2/3 of the project), but it happens each time. However, if I preview half my project separately, it usually don't freeze (so the project isn't corrupted). It seems to freeze when the preview rendering takes too long. Haven't try to render the project at the moment.

**Device Information:**
 - OS: Windows 10 Pro
 - OpenToonz Version: 1.5 stable
 - GPU: NVIDIA GTX 660
 - GPU Driver Version: 456.71
 - 16 Go RAM

**Comment by RodneyBaker** [comment]
What are you rendering to?  MP4?  Image sequences?
You might wish to render to image sequence first and them collect those images into a video format later.

Also, since you have so many frames to render you might want to take advantage of task/batch rendering.
If not familiar with that we can walk you through that process.

**Comment by Tartarin8** [comment]
Yes, I was talking about preview. I know I'm probably not using OT in the way it was designed to, in many ways, which probably is the source of some of my workflow problems. In fact I'm doing an entire song animation in one unique OT project, so you may find it's a long project.
Yes, I use some effects. The rendering is long and I haven't finished my animation yet, so I haven't tried to render it. Would you like me to try next night?

**Comment by Tartarin8** [comment]
I just did a rendering with my preview cam (low-fi), which is much faster than the final cam. It worked well. So the bug is only when previewing.

**Comment by konero** [comment]
Hmm, that's about 81 seconds and that's extremely long for a cut in animation, you should find ways during the storyboard stage to make good camera cuts and split those into separate scene files. Is your system running out of memory? I tested with a black circle and a blur FX for 500 frames and my system used 3.8GB just to preview it.

**Comment by Tartarin8** [comment]
I've just tested the preview with the task manager open. When OT freezes, it uses 2149 Mo. With my other programs open, Windows says I'm at 50% ram usage. So no, it doesn't seem to be a memory problem.
I was hoping to not use another software for the final video editing and render it with OT. I'm almost finished with my animation, it's a unique OT project. With some camera movements and transition effects, it looks like several scenes ; I think it's fine.
The preview problem isn't dramatic, after all, since I can preview some parts and I can render a file without problem. But... it's a bug, it seems.
